src/org/zaproxy/zap/extension/alertFilters/AlertFilterAPI.java

```diff
@@ -0,0 +1,168 @@
+
+	private static final Logger log = Logger.getLogger(AlertFilterAPI.class);
+
+	private static final String PREFIX = "alertFilter";
+
+	private static final String VIEW_ALERT_FILTER_LIST = "alertFilterList";
+
+	private static final String ACTION_ADD_ALERT_FILTER = "addAlertFilter";
+	private static final String ACTION_REMOVE_ALERT_FILTER = "removeAlertFilter";
+
+	private static final String PARAM_CONTEXT_ID = "contextId";
+	private static final String PARAM_RULE_ID = "ruleId";
+	private static final String PARAM_NEW_LEVEL = "newLevel";
+	private static final String PARAM_URL = "url";
+	private static final String PARAM_URL_IS_REGEX = "urlIsRegex";
+	private static final String PARAM_PARAMETER = "parameter";
+	private static final String PARAM_ENABLED = "enabled";
+
+	private ExtensionAlertFilters extension;
+
+	public AlertFilterAPI(ExtensionAlertFilters extension) {
+		super();
+		this.extension = extension;
+
+		this.addApiView(new ApiView(VIEW_ALERT_FILTER_LIST, null, new String[] { PARAM_CONTEXT_ID }));
+
+		this.addApiAction(new ApiAction(ACTION_ADD_ALERT_FILTER, 
+				new String[] { PARAM_CONTEXT_ID, PARAM_RULE_ID, PARAM_NEW_LEVEL },
+				new String[] { PARAM_URL, PARAM_URL_IS_REGEX, PARAM_PARAMETER, PARAM_ENABLED}));
+		this.addApiAction(new ApiAction(ACTION_REMOVE_ALERT_FILTER, 
+				new String[] { PARAM_CONTEXT_ID, PARAM_RULE_ID, PARAM_NEW_LEVEL },
+				new String[] { PARAM_URL, PARAM_URL_IS_REGEX, PARAM_PARAMETER, PARAM_ENABLED}));
+
+	}
+
+	@Override
+	public String getPrefix() {
+		return PREFIX;
+	}
+
+	@Override
+	public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+		log.debug("handleApiView " + name + " " + params.toString());
+		Context context;
+
+		switch (name) {
+		case VIEW_ALERT_FILTER_LIST:
+			ApiResponseList listResponse = new ApiResponseList(name);
+			context = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID);
+			List<AlertFilter> afs = extension.getContextAlertFilterManager(
+					context.getIndex()).getAlertFilters();
+			
+			for (AlertFilter af : afs) {
+				listResponse.addItem(buildResponseFromAlertFilter(af));
+			}
+			return listResponse;
+
+		default:
+			throw new ApiException(ApiException.Type.BAD_VIEW);
+		}
+	}
+
+	@Override
+	public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+		log.debug("handleApiAction " + name + " " + params.toString());
+
+		AlertFilter af; 
+		Context context;
+		switch (name) {
+		case ACTION_ADD_ALERT_FILTER:
+			context = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID);
+			af = new AlertFilter(context.getIndex(), 
+					ApiUtils.getIntParam(params, PARAM_RULE_ID), 
+					ApiUtils.getIntParam(params, PARAM_NEW_LEVEL), 
+					ApiUtils.getOptionalStringParam(params, PARAM_URL),
+					getParam(params, PARAM_URL_IS_REGEX, false),
+					ApiUtils.getOptionalStringParam(params, PARAM_PARAMETER),
+					getParam(params, PARAM_ENABLED, true));
+			
+			// TODO more validation, esp url!
```

Comment:
The ID of the rule should be validated or the UI updated to handle missing/unknown IDs, currently it throws a NPE when trying to edit/save a rule that has a made up ID. 
It would be great if it also validated the the new risk level.